source/dub/internal/vibecompat/data/json.d

```diff
@@ -1779,24 +1779,76 @@ void writeJsonString(R, bool pretty = false)(ref R dst, in Json json, size_t lev
 		case Json.Type.object:
 			dst.put('{');
 			bool first = true;
-			foreach( string k, ref const Json e; json ){
-				if( e.type == Json.Type.undefined ) continue;
-				if( !first ) dst.put(',');
-				first = false;
-				static if (pretty) {
+
+			static if (pretty) {
+				import std.algorithm.searching : canFind;
+				string[] keyOrder = [//determines the sorting order of the json
+					"name",
+					"description",
+					"homepage",
+					"authors",
+					"copyright",
+					"license",
+					"dependencies",
+					"systemDependencies",
+					"subPackages",
+					"configurations",
+					"buildTypes",
+					"-ddoxFilterArgs",
+					"targetType",
+					"targetName",
+					"targetPath",
+					"workingDirectory",
+					"subConfigurations",
+					"buildRequirements",
+					"buildOptions",
+					"libs",
+					"sourceFiles",
+					"sourcePaths",
+					"excludedSourceFiles",
+					"mainSourceFile",
+					"copyFiles",
+					"versions",
+					"debugVersions",
+					"importPaths",
+					"stringImportPaths",
+					"preGenerateCommands",
+					"postGenerateCommands",
+					"preBuildCommands",
+					"postBuildCommands",
+					"dflags",
+					"lflags"
```

Comment:
Is there any particular reason for this order? e.g. why not alphabetic? could take advantage of its sortedness then.

Reply:
No, there is not a specific reason for this order. It's just the order that I copied it down. 

Initially, I made it [sort alphabetically](https://github.com/dlang/dub/pull/1593/commits/a9e5dc1561aca03647e0cf434a26f9148d5f01a3), but changed it because @ZombineDev said it would make more sense if `name` came before `dependencies`.

I still think alphabetical order would be a better choice though. That way, the code wouldn't have to be updated each time there is a new JSON property is added (like happened this morning, in #1582). What are your ideas on this?

Reply:
I wonder whether it makes sense to sort the fields by reading the fields of structure PackageDescription. https://github.com/andre2007/dub/blob/0719452de25c93e8c590b01e08ec5b938f58a582/source/dub/description.d#L65

This avoids the problem of having multiple places to be adapted when new fields are introduced.

Reply:
@andre2007 I don't think that would work, because the order of the fields in a struct have no semantic value. After parsing the fields from `PackageDescription`, we would still have to sort them in some way.

@ZombineDev does it really even matter what order the elements are in, as long as they are consistent?

Reply:
I've reverted to the alphabetical order sorting, because that seems like the most efficient way to sort the fields.

Reply:
Thanks, I'm fine with whatever order really, as long as there is reason for it and if its not immediately obvious what the order is there is some documentation saying what it is and why.